scipy/spatial/spherical_voronoi.py

```diff
@@ -0,0 +1,416 @@
+    num_tetrahedra = tetrahedron_coord_array.shape[0]
+    #reshape the tetrahedron_coord_array to place all tetrahedra consecutively without nesting
+    tetrahedron_coord_array = np.reshape(tetrahedron_coord_array, (tetrahedron_coord_array.shape[0] * tetrahedron_coord_array.shape[1], tetrahedron_coord_array.shape[2]))
+    array_stacked_a_matrices = np.hstack((tetrahedron_coord_array, np.ones((num_tetrahedra * 4, 1))))
+    first_column_array_determinant_arrays = tetrahedron_coord_array[...,0] ** 2 + tetrahedron_coord_array[...,1] ** 2 + tetrahedron_coord_array[...,2] ** 2
+    first_column_array_determinant_arrays = first_column_array_determinant_arrays[:,np.newaxis]
+    final_column_array_determinant_arrays = np.ones((first_column_array_determinant_arrays.shape[0],1))
+    array_D_x_contents_before_determinant_calculation = np.hstack((first_column_array_determinant_arrays, tetrahedron_coord_array[...,1:],final_column_array_determinant_arrays))
+    array_middle_column_arrays_D_y = np.hstack((np.reshape(tetrahedron_coord_array[...,0], (tetrahedron_coord_array.shape[0],1)), np.reshape(tetrahedron_coord_array[...,2], (tetrahedron_coord_array.shape[0],1))))
+    array_D_y_contents_before_determinant_calculation = np.hstack((first_column_array_determinant_arrays, array_middle_column_arrays_D_y, final_column_array_determinant_arrays))
+    array_D_z_contents_before_determinant_calculation = np.hstack((first_column_array_determinant_arrays, tetrahedron_coord_array[...,:-1],final_column_array_determinant_arrays))
+    #split the arrays back to stacks of matrices
+    array_D_x_contents_before_determinant_calculation = np.array(np.split(array_D_x_contents_before_determinant_calculation, num_tetrahedra))
+    array_D_y_contents_before_determinant_calculation = np.array(np.split(array_D_y_contents_before_determinant_calculation, num_tetrahedra))
+    array_D_z_contents_before_determinant_calculation = np.array(np.split(array_D_z_contents_before_determinant_calculation, num_tetrahedra))
+    array_a_contents_before_determinant_calculation = np.array(np.split(array_stacked_a_matrices, num_tetrahedra))
+    #compute the determinants for the stacks of matrices assembled above
+    array_Dx_values = np.linalg.det(array_D_x_contents_before_determinant_calculation)
+    array_Dy_values = - np.linalg.det(array_D_y_contents_before_determinant_calculation)
+    array_Dz_values = np.linalg.det(array_D_z_contents_before_determinant_calculation)
+    array_a_values = np.linalg.det(array_a_contents_before_determinant_calculation)
+    array_denominator_values = 2. * array_a_values
+    array_x0_values = array_Dx_values / array_denominator_values
+    array_y0_values = array_Dy_values / array_denominator_values
+    array_z0_values = array_Dz_values / array_denominator_values
+    circumcenter_array = np.hstack((array_x0_values[:,np.newaxis], array_y0_values[:,np.newaxis], array_z0_values[:,np.newaxis]))
```

Comment:
Could simplify this to:

    circumcenter_array = np.column_stack((array_x0_values, array_y0_values, array_z0_values))